CHANGELOG.md

```diff
@@ -21,7 +21,31 @@ We appreciate your patience while we speedily work towards a stable release of t
 * Fixes an issue that could cause containers using `enable_memory_snapshot=True` on Python 3.9 and below to shut down prematurely
 
 
+### 0.64.97 (2024-09-11)
 
+* Adds support for [ASGI lifespan protocol](https://asgi.readthedocs.io/en/latest/specs/lifespan.html): 
+
+    ```python
+    @app.function()
+    @modal.asgi_app()
+    def func():
+        from fastapi import FastAPI, Request
+
+        def lifespan(wapp: FastAPI):
+            print("Starting")
+            yield {"foo": "bar"}
+            print("Shutting down")
+
+        web_app = FastAPI(lifespan=lifespan)
+
+        @web_app.get("/")
+        def get_state(request: Request):
+            return {"message": f"This is the state: {request.state.foo}"}
+
+        return web_app
+    ```
+
+    and enabling support for `gradio>=v4` amongst other libraries using lifespans
```

Comment:
I'm not sure if the false markup here is a problem or not but note that i think we have previously accidentally broken the svelte build by including `>` tokens in the changelog. Maybe the simplest change would be `gradio v4+`?

Reply:
Hm this works locally when building though but happy to change if you think it's safer
<img width="850" alt="Screenshot 2024-09-12 at 15 38 52" src="https://github.com/user-attachments/assets/671e7a5e-7f00-4f11-93f0-ad2eaf0ee4d2">

Reply:
ah that's good — maybe just an issue with `<` actually https://modal-com.slack.com/archives/C03130Y0AGL/p1722958342726519?thread_ts=1722956785.894109&cid=C03130Y0AGL

Reply:
Seems like it's fine because it's in a code block, so both < and > works if you stick them within ``